irr/src/CGLTFMeshFileLoader.cpp

```diff
@@ -0,0 +1,610 @@
+#include "CGLTFMeshFileLoader.h"
+
+#include "coreutil.h"
+#include "CSkinnedMesh.h"
+#include "ISkinnedMesh.h"
+#include "irrTypes.h"
+#include "IReadFile.h"
+#include "matrix4.h"
+#include "path.h"
+#include "S3DVertex.h"
+#include "quaternion.h"
+#include "vector3d.h"
+
+#include "tiniergltf.hpp"
+
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <limits>
+#include <memory>
+#include <optional>
+#include <stdexcept>
+#include <utility>
+#include <variant>
+#include <vector>
+
+/* Notes on the coordinate system.
+ *
+ * glTF uses a right-handed coordinate system where +Z is the
+ * front-facing axis, and Irrlicht uses a left-handed coordinate
+ * system where -Z is the front-facing axis.
+ * We convert between them by reflecting the mesh across the X axis.
```

Comment:
mirroring

Reply:
Isn't reflection the precise graphical term?

Reply:
both work, "mirroring" is probably slightly more intuitive / readable